src/components/plugin/Legend.jsx

```diff
@@ -7,21 +8,31 @@ import './styles/Legend.css'
 // Renders a legend for all map layers
 const Legend = ({ layers }) => {
     const [isOpen, toggleOpen] = useState(false)
```

Comment:
Since the setter receives the new value `setIsOpen` seems more appropriate as the fn name.